### PR TITLE
[Neo]: add benchmark that adopts existing tests.

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr adds benchmark to the neo project, currentlty the benchmark use the VM tests json.

## about the block benchmark:

block.txt contains the block and its start states, by loading the text, you can run the block as you run them in a complete ledger.

### format of the block.txt:

block.toarray
storagekey.id | storagekey.length | storagekey.Key | storageItem.length | storageitem.Value
...
...

### Why those blocks?

100 blocks that have the heavest IO, then the rest have the heavest computation.

Fixes # (issue)

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Existing tests passes.

**Test Configuration**: None

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Jim8y** [comment]
BenchmarkDotNet v0.13.12, macOS Sonoma 14.2.1 (23C71) [Darwin 23.2.0]
Apple M1 Pro, 1 CPU, 10 logical and 10 physical cores
.NET SDK 7.0.404
  [Host]     : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD


| Method                  | Mean        | Error    | StdDev   |
|------------------------ |------------:|---------:|---------:|
| TestOthers              |  2,611.5 us | 13.54 us | 12.00 us |
| TestOpCodesArrays       |  3,562.4 us |  9.69 us |  7.56 us |
| TestOpCodesStack        |  1,518.5 us |  3.94 us |  3.69 us |
| TestOpCodesSlot         |  2,151.1 us | 10.13 us |  8.98 us |
| TestOpCodesSplice       | 23,359.8 us | 42.08 us | 37.30 us |
| TestOpCodesControl      |  2,905.6 us | 11.11 us |  9.84 us |
| TestOpCodesPush         |    319.4 us |  2.07 us |  1.84 us |
| TestOpCodesArithmetic   |  2,815.1 us | 11.94 us | 11.17 us |
| TestOpCodesBitwiseLogic |  2,620.3 us | 17.66 us | 14.75 us |
| TestOpCodesTypes        |    480.9 us |  1.40 us |  1.24 us |

**Comment by Jim8y** [comment]
BenchmarkDotNet v0.13.12, macOS Sonoma 14.2.1 (23C71) [Darwin 23.2.0]
Apple M1 Pro, 1 CPU, 10 logical and 10 physical cores
.NET SDK 7.0.404
  [Host]     : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD


| Method        | Mean            | Error        | StdDev       |
|-------------- |----------------:|-------------:|-------------:|
| NeoIssue2528  | 1,433,521.63 us | 9,596.338 us | 8,013.377 us |
| NeoVMIssue418 |    63,883.54 us |   227.887 us |   202.016 us |
| NeoIssue2723  |        23.85 us |     0.135 us |     0.105 us |

**Comment by Jim8y** [comment]
@superboyiii you will not like to test this,it will take many many hours to run......

**Comment by cschuchardt88** [comment]
What do those numbers even mean? Are they bad, normal or good? A better test would be `CPU` and `Memory`. At least we know what is good and bad. What kind of load was put on the `VM`? These benchmarks shows nothing that we can work with. You need something to compare too. Maybe put up ethereum and other blockchain `VM` engines to see the different.

**Comment by Jim8y** [comment]
BenchmarkDotNet v0.13.12, macOS Sonoma 14.2.1 (23C71) [Darwin 23.2.0]
Apple M1 Pro, 1 CPU, 10 logical and 10 physical cores
.NET SDK 7.0.404
  [Host]     : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD
  DefaultJob : .NET 7.0.14 (7.0.1423.51910), Arm64 RyuJIT AdvSIMD


| Method                              | Mean          | Error      | StdDev     |
|------------------------------------ |--------------:|-----------:|-----------:|
| TestOthersDebugger                  |    167.315 us |  2.0578 us |  1.8242 us |
| TestOthersStackItemLimits           |    153.624 us |  2.6652 us |  2.7370 us |
| TestOthersScriptLogic               |     33.388 us |  0.5530 us |  0.7931 us |
| TestOthersOtherCases                |     24.263 us |  0.1436 us |  0.1344 us |
| TestOthersInvocationLimits          |    436.671 us |  3.3312 us |  2.7817 us |
| TestOthersInit                      |      7.833 us |  0.0333 us |  0.0278 us |
| TestOthersStackLimits               |  1,469.619 us |  3.2813 us |  2.9088 us |
| TestOpCodesArraysClearItems         |     73.256 us |  0.4625 us |  0.3611 us |
| TestOpCodesArraysPack               |    123.086 us |  2.2047 us |  3.6224 us |
| TestOpCodesArraysNewArrayT          |    100.790 us |  0.7284 us |  0.6457 us |
| TestOpCodesArraysPickItem           |    629.811 us |  3.2570 us |  2.8873 us |
| TestOpCodesArraysPackStruct         |    118.687 us |  0.3599 us |  0.3191 us |
| TestOpCodesArraysSetItem            |    741.747 us |  3.0782 us |  2.5705 us |
| TestOpCodesArraysNewStruct0         |      4.713 us |  0.0113 us |  0.0100 us |
| TestOpCodesArraysReverseItems       |     79.934 us |  0.4499 us |  0.3988 us |
| TestOpCodesArraysNewMap             |     13.464 us |  0.2547 us |  0.2502 us |
| TestOpCodesArraysHasKey             |    151.972 us |  0.5114 us |  0.4533 us |
| TestOpCodesArraysNewArray           |    123.620 us |  0.2284 us |  0.2025 us |
| TestOpCodesArraysKeys               |     38.109 us |  0.1210 us |  0.1010 us |
| TestOpCodesArraysAppend             |    375.020 us |  4.9101 us |  4.1002 us |
| TestOpCodesArraysRemove             |    186.428 us |  1.6746 us |  1.4845 us |
| TestOpCodesArraysPackMap            |    118.274 us |  0.4290 us |  0.4013 us |
| TestOpCodesArraysValues             |    114.051 us |  0.5210 us |  0.4619 us |
| TestOpCodesArraysNewArray0          |      4.782 us |  0.0077 us |  0.0072 us |
| TestOpCodesArraysUnpack             |     95.467 us |  0.3020 us |  0.2677 us |
| TestOpCodesArraysSize               |    113.755 us |  0.6225 us |  0.5822 us |
| TestOpCodesArraysNewStruct          |    124.042 us |  0.5843 us |  0.5179 us |
| TestOpCodesStackXDrop               |    153.261 us |  0.6088 us |  0.5083 us |
| TestOpCodesStackReverseN            |    113.980 us |  1.0531 us |  0.8794 us |
| TestOpCodesStackReverse4            |     48.869 us |  0.1053 us |  0.0985 us |
| TestOpCodesStackClear               |     12.588 us |  0.0268 us |  0.0251 us |
| TestOpCodesStackReverse3            |     44.376 us |  0.1216 us |  0.1078 us |
| TestOpCodesStackRot                 |    119.945 us |  0.1825 us |  0.1707 us |
| TestOpCodesStackPick                |    220.589 us |  0.6408 us |  0.5994 us |
| TestOpCodesStackNip                 |     75.735 us |  0.4691 us |  0.3917 us |
| TestOpCodesStackRoll                |    235.019 us |  0.8874 us |  0.7410 us |
| TestOpCodesStackDepth               |     78.397 us |  0.0818 us |  0.0725 us |
| TestOpCodesStackSwap                |    108.553 us |  0.3208 us |  0.3001 us |
| TestOpCodesStackTuck                |    122.958 us |  0.4718 us |  0.4413 us |
| TestOpCodesStackOver                |    124.121 us |  0.1959 us |  0.1529 us |
| TestOpCodesStackDrop                |     38.120 us |  0.0764 us |  0.0715 us |
| TestOpCodesSlotStsFld6              |     64.251 us |  0.2888 us |  0.2255 us |
| TestOpCodesSlotInitSlot             |    107.463 us |  0.3696 us |  0.3276 us |
| TestOpCodesSlotLdLoc4               |     47.437 us |  0.1149 us |  0.1075 us |
| TestOpCodesSlotLdArg                |     47.193 us |  0.0846 us |  0.0707 us |
| TestOpCodesSlotStArg0               |     23.759 us |  0.0691 us |  0.0613 us |
| TestOpCodesSlotLdSFLD0              |     26.362 us |  0.0775 us |  0.0687 us |
| TestOpCodesSlotLdArg4               |     48.179 us |  0.1415 us |  0.1255 us |
| TestOpCodesSlotLdArg5               |     48.425 us |  0.1131 us |  0.1058 us |
| TestOpCodesSlotLdSFLD1              |     47.141 us |  0.1554 us |  0.1377 us |
| TestOpCodesSlotLdLoc                |     47.164 us |  0.0747 us |  0.0624 us |
| TestOpCodesSlotStArg1               |     45.206 us |  0.0934 us |  0.0874 us |
| TestOpCodesSlotLdLoc5               |     47.346 us |  0.1939 us |  0.1719 us |
| TestOpCodesSlotLdSFLD6              |     47.458 us |  0.0877 us |  0.0821 us |
| TestOpCodesSlotLdArg2               |     47.654 us |  0.1345 us |  0.1258 us |
| TestOpCodesSlotStsFld0              |     32.215 us |  0.0598 us |  0.0530 us |
| TestOpCodesSlotLdLoc2               |     47.165 us |  0.0910 us |  0.0852 us |
| TestOpCodesSlotStArg6               |     46.379 us |  0.1623 us |  0.1439 us |
| TestOpCodesSlotLdLoc3               |     46.678 us |  0.0966 us |  0.0856 us |
| TestOpCodesSlotStsFld1              |     55.301 us |  0.0832 us |  0.0738 us |
| TestOpCodesSlotInitSSlot            |     59.818 us |  0.1898 us |  0.1585 us |
| TestOpCodesSlotLdArg3               |     48.029 us |  0.0931 us |  0.0825 us |
| TestOpCodesSlotLdSFLD               |     47.087 us |  0.1195 us |  0.1118 us |
| TestOpCodesSlotLdArg0               |     26.231 us |  0.0931 us |  0.0826 us |
| TestOpCodesSlotLdSFLD4              |     47.847 us |  0.0966 us |  0.0903 us |
| TestOpCodesSlotStArg4               |     45.506 us |  0.0449 us |  0.0398 us |
| TestOpCodesSlotLdLoc0               |     26.966 us |  0.2988 us |  0.2795 us |
| TestOpCodesSlotStsFld2              |     58.786 us |  0.4085 us |  0.3821 us |
| TestOpCodesSlotStsFld3              |     58.756 us |  0.1248 us |  0.1167 us |
| TestOpCodesSlotLdLoc1               |     47.486 us |  0.0715 us |  0.0634 us |
| TestOpCodesSlotStArg5               |     46.214 us |  0.1814 us |  0.1608 us |
| TestOpCodesSlotLdSFLD5              |     47.503 us |  0.1003 us |  0.0938 us |
| TestOpCodesSlotLdArg1               |     47.577 us |  0.1040 us |  0.0922 us |
| TestOpCodesSlotStArg2               |     45.221 us |  0.1238 us |  0.1097 us |
| TestOpCodesSlotLdLoc6               |     47.225 us |  0.0768 us |  0.0718 us |
| TestOpCodesSlotStArg                |     44.959 us |  0.1054 us |  0.0880 us |
| TestOpCodesSlotStsFld4              |     60.535 us |  0.1612 us |  0.1508 us |
| TestOpCodesSlotLdArg6               |     48.804 us |  0.1106 us |  0.1034 us |
| TestOpCodesSlotLdSFLD2              |     47.071 us |  0.1968 us |  0.1745 us |
| TestOpCodesSlotLdSFLD3              |     47.657 us |  0.1297 us |  0.1150 us |
| TestOpCodesSlotStLoc                |     56.065 us |  0.1303 us |  0.1155 us |
| TestOpCodesSlotStsFld               |     53.165 us |  0.1147 us |  0.1017 us |
| TestOpCodesSlotStsFld5              |     62.397 us |  0.1189 us |  0.1112 us |
| TestOpCodesSlotStArg3               |     45.435 us |  0.1251 us |  0.1109 us |
| TestOpCodesSpliceSubStr             |  1,152.939 us |  2.5335 us |  1.9780 us |
| TestOpCodesSpliceCat                | 21,459.463 us | 68.9387 us | 64.4853 us |
| TestOpCodesSpliceMemCpy             |    235.761 us |  0.5563 us |  0.4931 us |
| TestOpCodesSpliceLeft               |    145.490 us |  0.6669 us |  0.5912 us |
| TestOpCodesSpliceRight              |    171.344 us |  0.2651 us |  0.2350 us |
| TestOpCodesSpliceNewBuffer          |    117.655 us |  0.2880 us |  0.2553 us |
| TestOpCodesControlTryCatchFinally6  |     20.269 us |  0.1034 us |  0.0967 us |
| TestOpCodesControlJmpLeL            |    129.313 us |  0.2893 us |  0.2565 us |
| TestOpCodesControlJmpEqL            |    134.667 us |  0.2003 us |  0.1776 us |
| TestOpCodesControlJmpLe             |    129.427 us |  0.1888 us |  0.1674 us |
| TestOpCodesControlJmpNe             |    128.853 us |  0.1686 us |  0.1494 us |
| TestOpCodesControlTryCatchFinally7  |     29.819 us |  0.0671 us |  0.0595 us |
| TestOpCodesControlAssertMsg         |     24.470 us |  0.0473 us |  0.0443 us |
| TestOpCodesControlJmpGt             |    134.766 us |  0.2772 us |  0.2314 us |
| TestOpCodesControlJmpL              |     50.252 us |  0.2671 us |  0.2498 us |
| TestOpCodesControlJmpIfL            |     60.140 us |  0.0999 us |  0.0934 us |
| TestOpCodesControlTryFinally        |     27.130 us |  0.0373 us |  0.0331 us |
| TestOpCodesControlJmpNeL            |    128.761 us |  0.1945 us |  0.1724 us |
| TestOpCodesControlJmpIfNot          |     59.602 us |  0.1227 us |  0.1025 us |
| TestOpCodesControlAbortMsg          |     21.255 us |  0.0514 us |  0.0455 us |
| TestOpCodesControlCall              |     80.315 us |  0.1491 us |  0.1322 us |
| TestOpCodesControlCallL             |     80.644 us |  0.1569 us |  0.1391 us |
| TestOpCodesControlJmpGeL            |    128.862 us |  0.2178 us |  0.1819 us |
| TestOpCodesControlRet               |      2.102 us |  0.0016 us |  0.0015 us |
| TestOpCodesControlJmpGtL            |    135.377 us |  0.3080 us |  0.2730 us |
| TestOpCodesControlJmpEq             |    134.651 us |  0.1383 us |  0.1155 us |
| TestOpCodesControlSysCall           |    115.962 us |  0.1331 us |  0.1111 us |
| TestOpCodesControlCallA             |     75.181 us |  0.1436 us |  0.1273 us |
| TestOpCodesControlAssert            |     23.130 us |  0.0399 us |  0.0353 us |
| TestOpCodesControlTryCatchFinally2  |     32.400 us |  0.0198 us |  0.0185 us |
| TestOpCodesControlTryCatchFinally3  |     33.430 us |  0.1189 us |  0.0993 us |
| TestOpCodesControlNop               |     30.171 us |  0.0225 us |  0.0211 us |
| TestOpCodesControlJmpGe             |    130.177 us |  2.3441 us |  2.0780 us |
| TestOpCodesControlTryCatchFinally10 |     20.753 us |  0.1032 us |  0.0965 us |
| TestOpCodesControlJmpLt             |    139.097 us |  2.2198 us |  2.0764 us |
| TestOpCodesControlJmpIf             |     59.955 us |  0.1005 us |  0.0940 us |
| TestOpCodesControlAbort             |     20.231 us |  0.0730 us |  0.0647 us |
| TestOpCodesControlJmpIfNotL         |     60.286 us |  0.1776 us |  0.1661 us |
| TestOpCodesControlTryCatch          |     49.312 us |  0.0866 us |  0.0723 us |
| TestOpCodesControlTryCatchFinally4  |     35.640 us |  0.0574 us |  0.0508 us |
| TestOpCodesControlJmp               |     49.796 us |  0.1959 us |  0.1737 us |
| TestOpCodesControlTryCatchFinally8  |     49.559 us |  0.0959 us |  0.0897 us |
| TestOpCodesControlTryCatchFinally9  |     29.632 us |  0.0544 us |  0.0454 us |
| TestOpCodesControlJmpLtL            |    135.380 us |  0.1195 us |  0.0998 us |
| TestOpCodesControlTryCatchFinally   |     19.597 us |  0.0291 us |  0.0258 us |
| TestOpCodesControlTryCatchFinally5  |     20.545 us |  0.0250 us |  0.0208 us |
| TestOpCodesControlThrow             |     20.591 us |  0.0344 us |  0.0288 us |
| TestOpCodesPushPushA                |     49.620 us |  0.0694 us |  0.0615 us |
| TestOpCodesPushPushM1ToPush16       |     92.773 us |  0.1055 us |  0.0881 us |
| TestOpCodesPushPushInt8ToPushInt256 |     16.794 us |  0.0222 us |  0.0207 us |
| TestOpCodesPushPushData1            |     26.316 us |  0.0846 us |  0.0750 us |
| TestOpCodesPushPushData2            |     26.231 us |  0.0225 us |  0.0199 us |
| TestOpCodesPushPushNull             |     11.430 us |  0.0089 us |  0.0079 us |
| TestOpCodesPushPushData4            |     88.262 us |  0.1580 us |  0.1478 us |
| TestOpCodesArithmeticGe             |    178.903 us |  0.2013 us |  0.1785 us |
| TestOpCodesArithmeticLt             |    178.600 us |  0.2506 us |  0.2222 us |
| TestOpCodesArithmeticModMul         |     58.140 us |  0.1173 us |  0.1040 us |
| TestOpCodesArithmeticNumNotEqual    |    542.206 us |  1.4754 us |  1.2320 us |
| TestOpCodesArithmeticNot            |    192.275 us |  0.3539 us |  0.2955 us |
| TestOpCodesArithmeticModPow         |     58.464 us |  0.1312 us |  0.1095 us |
| TestOpCodesArithmeticLe             |    178.272 us |  0.3116 us |  0.2763 us |
| TestOpCodesArithmeticShl            |    113.378 us |  0.2234 us |  0.1866 us |
| TestOpCodesArithmeticGt             |    188.743 us |  0.3583 us |  0.3176 us |
| TestOpCodesArithmeticPow            |     82.798 us |  0.0822 us |  0.0642 us |
| TestOpCodesArithmeticNumEqual       |    542.583 us |  3.4831 us |  3.0876 us |
| TestOpCodesArithmeticSign           |    250.366 us |  0.4581 us |  0.3577 us |
| TestOpCodesArithmeticSqrt           |     70.828 us |  0.0706 us |  0.0660 us |
| TestOpCodesArithmeticShr            |    113.920 us |  0.1127 us |  0.0941 us |
| TestOpCodesBitwiseLogicOr           |    405.847 us |  0.6763 us |  0.5647 us |
| TestOpCodesBitwiseLogicEqual        |    530.956 us |  1.2476 us |  1.0418 us |
| TestOpCodesBitwiseLogicInvert       |    325.254 us |  0.5096 us |  0.4518 us |
| TestOpCodesBitwiseLogicXor          |    404.443 us |  1.2724 us |  1.1280 us |
| TestOpCodesBitwiseLogicNotEqual     |    483.648 us |  1.2789 us |  1.1337 us |
| TestOpCodesBitwiseLogicAnd          |    402.503 us |  0.3831 us |  0.3583 us |
| TestOpCodesTypesConvert             |    368.866 us |  0.5782 us |  0.5126 us |
| TestOpCodesTypesIsType              |     43.451 us |  0.1707 us |  0.1513 us |
| TestOpCodesTypesIsNull              |     63.196 us |  0.0939 us |  0.0879 us |

**Comment by Jim8y** [comment]
> What do those numbers even mean? Are they bad, normal or good? A better test would be `CPU` and `Memory`. At least we know what is good and bad. What kind of load was put on the `VM`? These benchmarks shows nothing that we can work with. You need something to compare too. Maybe put up ethereum and other blockchain `VM` engines to see the different. Or compare to the `NEO` 2 or the old `NEO` blockchain.

Have it first, improve it later. If this hurt you, i am sorry, but i need something to see if my optimize really works. I will run benchmark after i change my code and compare the difference. But i dont understand why you even have problem with `Benchmarkdotnet`. You dont like it, you just dont run it.

**Comment by cschuchardt88** [comment]
Its doesn't hurt. You just provide no information other than an output. What the sample rate? Also running someone on one machine doesn't show anything either. You run the benchmark in a `headless` `vm` like a github action or your own environment with nothing running other this test. A `docker` container would be best.

**Comment by Jim8y** [comment]
> Its doesn't hurt. You just provide no information other than an output. What's the sample rate? Also running something on one machine doesn't show anything either. You run the benchmark in a `headless` `vm` like a github action or your own environment with nothing running other this test. A `docker` container would be best.

This benchmark is not for showing people `how fast and how efficient` it is, but for us developers (yes, me) to compare the performance difference after we make change to the core, in the same machine, same environment with high priority.

We need to know how much we improved.

**Comment by shargon** [comment]
> > What do those numbers even mean? Are they bad, normal or good? A better test would be `CPU` and `Memory`. At least we know what is good and bad. What kind of load was put on the `VM`? These benchmarks shows nothing that we can work with. You need something to compare too. Maybe put up ethereum and other blockchain `VM` engines to see the different. Or compare to the `NEO` 2 or the old `NEO` blockchain.
> 
> **Have it first, improve it later.** If this hurt you, i am sorry, but i need something to see if my optimization really works, if my later code update improve the benchmark or not. I will run benchmark after i change my code and compare the difference.

Totally agree

File: benchmarks/Neo.VM.Benchmarks/Benchmark.OpCodes.cs [hunk 1]
```diff
@@ -0,0 +1,867 @@
...
+
+        [Benchmark]
+        public void TestOpCodesControlAssertMsg() => RunTest(_opCodesControlAssertMsg);
+
+        [Benchmark]
+        public void TestOpCodesControlJmpGt() => RunTest(_opCodesControlJmpGt);
+
+        [Benchmark]
+
+        public void TestOpCodesControlJmpL() => RunTest(_opCodesControlJmpL);
```
Comment (shargon) [inline benchmarks/Neo.VM.Benchmarks/Benchmark.OpCodes.cs:652]:
```suggestion
        [Benchmark]
        public void TestOpCodesControlJmpL() => RunTest(_opCodesControlJmpL);
```

**Review by shargon — COMMENTED** [review]
One more

**Review by shargon — DISMISSED** [review]

**Comment by Jim8y** [comment]
| Method   | _blockId | Mean         | Error        | StdDev       | Median       |
|--------- |--------- |-------------:|-------------:|-------------:|-------------:|
| RunBlock | 470619   | 59,874.25 us |   656.686 us |   614.264 us | 59,664.97 us |
| RunBlock | 715707   | 59,676.45 us | 1,026.906 us |   801.740 us | 59,548.54 us |
| RunBlock | 1466600  |    702.38 us |    11.482 us |    16.468 us |    697.37 us |
| RunBlock | 1545151  |  3,629.55 us |    66.468 us |    62.175 us |  3,646.19 us |
| RunBlock | 1551539  | 40,426.02 us |   296.153 us |   262.532 us | 40,518.37 us |
| RunBlock | 1551550  | 37,132.35 us |   595.948 us |   497.643 us | 37,016.54 us |
| RunBlock | 1551551  | 36,716.81 us |   436.687 us |   387.112 us | 36,727.37 us |
| RunBlock | 1551552  | 72,477.42 us |   449.472 us |   398.445 us | 72,449.76 us |
| RunBlock | 1570420  |  3,872.30 us |    56.277 us |    46.994 us |  3,868.17 us |
| RunBlock | 1584313  | 63,325.32 us | 1,223.393 us | 2,044.014 us | 63,055.17 us |
| RunBlock | 1667870  | 47,243.42 us |   932.744 us | 1,841.145 us | 46,414.82 us |
| RunBlock | 1667871  | 45,389.34 us |   418.586 us |   349.538 us | 45,278.60 us |
| RunBlock | 1683031  |    695.25 us |     3.551 us |     2.772 us |    694.95 us |
| RunBlock | 1690032  |  4,429.23 us |    20.914 us |    17.464 us |  4,427.75 us |
| RunBlock | 1690810  |  8,128.87 us |   160.873 us |   339.336 us |  8,032.39 us |
| RunBlock | 1691179  | 63,397.90 us | 1,225.681 us | 1,550.094 us | 62,643.42 us |
| RunBlock | 1691180  | 45,237.77 us |   233.863 us |   207.314 us | 45,219.09 us |
| RunBlock | 1691181  | 28,954.83 us |   387.026 us |   343.089 us | 28,906.32 us |
| RunBlock | 1691492  |  5,070.97 us |    71.992 us |    63.819 us |  5,038.64 us |
| RunBlock | 1691562  |  3,365.98 us |    54.508 us |    48.320 us |  3,347.82 us |
| RunBlock | 1691721  |  5,325.29 us |    12.088 us |     9.438 us |  5,325.14 us |
| RunBlock | 1691783  |  5,070.58 us |   100.999 us |   120.232 us |  5,080.86 us |
| RunBlock | 1691784  |  5,757.11 us |    93.432 us |    82.825 us |  5,736.62 us |
| RunBlock | 1691811  |  7,606.66 us |    78.561 us |    61.336 us |  7,599.67 us |
| RunBlock | 1691968  |  3,169.93 us |    16.585 us |    13.849 us |  3,167.62 us |
| RunBlock | 1692023  |  6,952.28 us |    34.964 us |    29.197 us |  6,937.13 us |
| RunBlock | 1692102  |  8,451.20 us |    51.944 us |    43.376 us |  8,438.83 us |
| RunBlock | 1692302  |  3,970.65 us |    77.830 us |   106.534 us |  3,918.85 us |
| RunBlock | 1692680  |  2,981.37 us |    51.223 us |    42.773 us |  2,970.88 us |
| RunBlock | 1692682  |  6,234.25 us |   124.196 us |   185.891 us |  6,193.75 us |
| RunBlock | 1692692  |  3,697.42 us |    72.447 us |    64.222 us |  3,665.42 us |
| RunBlock | 1692735  |  3,564.20 us |    26.134 us |    21.823 us |  3,554.92 us |
| RunBlock | 1692772  |  4,990.38 us |    47.416 us |    39.594 us |  4,993.74 us |
| RunBlock | 1692921  |  4,648.66 us |    92.645 us |    95.139 us |  4,648.51 us |
| RunBlock | 1693099  |  4,333.95 us |    66.502 us |    62.206 us |  4,313.34 us |
| RunBlock | 1693100  |  3,121.52 us |    43.932 us |    36.685 us |  3,119.83 us |
| RunBlock | 1693182  | 12,591.30 us |   250.557 us |   307.707 us | 12,415.09 us |
| RunBlock | 1693194  |  3,999.73 us |    40.606 us |    35.996 us |  4,012.20 us |
| RunBlock | 1693241  |  5,904.57 us |    16.436 us |    13.725 us |  5,902.75 us |
| RunBlock | 1693269  |  9,304.86 us |    96.158 us |    75.074 us |  9,271.72 us |
| RunBlock | 1693274  |  7,921.25 us |    41.556 us |    36.838 us |  7,914.82 us |
| RunBlock | 1693286  |  6,912.90 us |    46.279 us |    38.645 us |  6,900.75 us |
| RunBlock | 1693294  |  9,654.82 us |    41.611 us |    36.887 us |  9,657.01 us |
| RunBlock | 1693308  |  5,307.60 us |   106.040 us |   113.461 us |  5,309.95 us |
| RunBlock | 1693343  |  5,995.49 us |   106.332 us |    99.463 us |  5,978.68 us |
| RunBlock | 1693373  |  2,693.75 us |    42.853 us |    40.085 us |  2,684.11 us |
| RunBlock | 1693398  |  4,324.42 us |    85.516 us |    91.502 us |  4,310.20 us |
| RunBlock | 1693399  |  3,667.85 us |    56.037 us |    52.418 us |  3,676.16 us |
| RunBlock | 1693428  |  4,305.41 us |    84.770 us |   104.106 us |  4,279.38 us |
| RunBlock | 1693650  |  4,049.93 us |    60.492 us |    50.514 us |  4,047.15 us |
| RunBlock | 1694343  |  4,834.97 us |    96.302 us |   144.140 us |  4,840.81 us |
| RunBlock | 1694410  |  4,583.50 us |    13.212 us |    12.358 us |  4,579.54 us |
| RunBlock | 1694594  |  2,096.84 us |    11.531 us |    10.786 us |  2,098.16 us |
| RunBlock | 1695163  |  3,736.47 us |    13.725 us |    12.167 us |  3,737.18 us |
| RunBlock | 1695746  |  3,563.30 us |    69.569 us |   128.951 us |  3,583.94 us |
| RunBlock | 1695896  |  8,624.72 us |   170.171 us |   221.271 us |  8,623.32 us |
| RunBlock | 1696740  |  3,540.13 us |    66.403 us |    62.113 us |  3,534.77 us |
| RunBlock | 1696758  | 54,196.75 us |   837.028 us |   742.003 us | 54,153.36 us |
| RunBlock | 1696759  | 71,179.58 us | 1,025.331 us | 1,007.012 us | 71,119.91 us |
| RunBlock | 1723984  |  4,695.97 us |    31.997 us |    28.365 us |  4,696.30 us |
| RunBlock | 1855694  |  3,740.07 us |    62.696 us |    58.646 us |  3,744.03 us |
| RunBlock | 1855827  |  4,452.93 us |    60.198 us |    56.309 us |  4,475.39 us |
| RunBlock | 1882087  |  8,181.28 us |   128.148 us |   119.870 us |  8,179.60 us |
| RunBlock | 1882106  | 10,445.59 us |    74.218 us |    69.424 us | 10,447.23 us |
| RunBlock | 1899635  |  1,785.30 us |    21.255 us |    19.882 us |  1,792.88 us |
| RunBlock | 1900762  |  7,205.54 us |    74.863 us |    58.448 us |  7,223.80 us |
| RunBlock | 1900767  |  5,293.33 us |    32.736 us |    29.020 us |  5,293.10 us |
| RunBlock | 1900768  |  5,024.04 us |    39.675 us |    35.170 us |  5,031.96 us |
| RunBlock | 1900845  |  6,000.28 us |    72.079 us |    63.896 us |  5,981.22 us |
| RunBlock | 1907366  |  5,843.61 us |   105.216 us |    98.419 us |  5,850.23 us |
| RunBlock | 1941906  |  7,338.10 us |    93.257 us |    87.232 us |  7,332.16 us |
| RunBlock | 1952842  |  6,732.85 us |    81.065 us |    75.828 us |  6,720.66 us |
| RunBlock | 1955515  |  7,177.34 us |    56.432 us |    47.123 us |  7,164.92 us |
| RunBlock | 1976901  |     94.04 us |     1.120 us |     1.047 us |     94.00 us |
| RunBlock | 1991342  |  2,543.55 us |    47.418 us |    44.355 us |  2,532.16 us |
| RunBlock | 2027305  |    489.78 us |     3.146 us |     2.627 us |    490.75 us |
| RunBlock | 2037494  |  2,568.57 us |    48.806 us |    45.653 us |  2,564.36 us |
| RunBlock | 2080810  |  3,675.61 us |    57.580 us |    80.719 us |  3,629.25 us |
| RunBlock | 2116627  |    527.96 us |     1.071 us |     0.836 us |    527.86 us |
| RunBlock | 2137993  |  1,076.29 us |     9.755 us |     9.125 us |  1,075.34 us |
| RunBlock | 2141425  |  1,386.99 us |    19.729 us |    18.455 us |  1,379.95 us |
| RunBlock | 2141551  |  2,428.17 us |    47.977 us |    94.701 us |  2,405.82 us |
| RunBlock | 2158835  |  1,331.44 us |     1.481 us |     1.157 us |  1,331.67 us |
| RunBlock | 2171359  |    986.42 us |     6.510 us |     5.771 us |    987.77 us |
| RunBlock | 2194622  |  1,258.00 us |     4.145 us |     3.461 us |  1,258.21 us |
| RunBlock | 2212242  | 25,737.63 us |   500.592 us |   576.482 us | 25,747.40 us |
| RunBlock | 2212278  | 22,415.22 us |   158.055 us |   131.983 us | 22,389.10 us |
| RunBlock | 2212440  | 24,013.78 us |   361.602 us |   338.243 us | 24,115.69 us |
| RunBlock | 2212544  | 33,361.35 us |   336.353 us |   314.625 us | 33,351.93 us |
| RunBlock | 2212575  | 36,464.44 us |   433.404 us |   384.201 us | 36,531.90 us |
| RunBlock | 2212681  | 46,199.06 us |   770.673 us |   683.182 us | 46,212.38 us |
| RunBlock | 2212706  | 48,617.33 us |   968.430 us | 1,746.279 us | 48,388.32 us |
| RunBlock | 2212723  | 23,246.17 us |   244.452 us |   228.660 us | 23,273.53 us |
| RunBlock | 2212800  | 23,901.17 us |   466.449 us |   589.909 us | 23,870.32 us |
| RunBlock | 2212807  | 23,733.14 us |   332.395 us |   310.923 us | 23,803.55 us |
| RunBlock | 2212833  | 24,021.51 us |   226.906 us |   189.476 us | 23,993.33 us |
| RunBlock | 2230072  |  1,336.33 us |     7.872 us |     6.146 us |  1,336.24 us |
| RunBlock | 2268049  |    859.48 us |     5.150 us |     4.565 us |    859.24 us |
| RunBlock | 2351614  |    767.84 us |    15.060 us |    16.739 us |    766.19 us |
| RunBlock | 2363914  |  2,029.60 us |    38.871 us |    59.360 us |  2,017.69 us |

**Comment by shargon** [comment]
conflicts

**Pull request closed without merging**